### ISSUE TITLE
Dirty reactor when running test_download_torrent_from_url

### ISSUE THREAD

**Issue by whirm** [Tribler/tribler#2147]
```
DEBUG   1461854028  LaunchManyCore:TriblerLaunchMany:721  tlm: network checkpointing: to file /home/jenkins/workspace/GH_Tribler_PR_tests_linux/tmp/tmpBqThSp_tribler_test_session/dot.Tribler/dlcheckpoints/fc8a15a2faf2734dbb1dc5f7afdc5c9beaeb1f59.state
INFO    1461854028  LaunchManyCore:TriblerLaunchMany:575  tlm: shutdown: delaying for early shutdown tasks 1.90339612961
DEBUG   1461854029 LibtorrentDownloadImpl:LibtorrentDownloadImpl:743  Torrent ubuntu-15.04-desktop-amd64.iso PROGRESS 0.0 DLSTATE 5 SEEDTIME 0
DEBUG   1461854029   DownloadState:DownloadState:64   we have status and it is not downloading or seeding 'ubuntu-15.04-desktop-amd64.iso'
DEBUG   1461854029   test_download:TestDownload:70   download status: 'ubuntu-15.04-desktop-amd64.iso' DLSTATUS_STOPPED 0.0
DEBUG   1461854029  test_as_server:TestDownload:359  Waiting for Session to shutdown, will wait for an additional 58 seconds
DEBUG   1461854030 LibtorrentDownloadImpl:LibtorrentDownloadImpl:743  Torrent ubuntu-15.04-desktop-amd64.iso PROGRESS 0.0 DLSTATE 5 SEEDTIME 0
DEBUG   1461854030   DownloadState:DownloadState:64   we have status and it is not downloading or seeding 'ubuntu-15.04-desktop-amd64.iso'
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:695  tlm: network_shutdown
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:698  tlm: Number of threads still running 4
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:700  tlm: Thread still running=MainThread, daemon=False, instance=<_MainThread(MainThread, started 140133437900544)>
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:700  tlm: Thread still running=Twisted, daemon=True, instance=<Thread(Twisted, started daemon 140133267416832)>
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:700  tlm: Thread still running=PoolThread-twisted.internet.reactor-0, daemon=True, instance=<Thread(PoolThread-twisted.internet.reactor-0, started daemon 140132913927936)>
INFO    1461854030  LaunchManyCore:TriblerLaunchMany:700  tlm: Thread still running=WatchDog, daemon=True, instance=<WatchDog(WatchDog, started daemon 140132961994496)>
DEBUG   1461854030   test_download:TestDownload:70   download status: 'ubuntu-15.04-desktop-amd64.iso' DLSTATUS_STOPPED 0.0
DEBUG   1461854030  test_as_server:TestDownload:364  Session has shut down
DEBUG   1461854030  test_as_server:TestDownload:276  test_as_server: Number of threads still running 4
DEBUG   1461854030  test_as_server:TestDownload:278  Thread still running MainThread, daemon: False, instance: <_MainThread(MainThread, started 140133437900544)>
DEBUG   1461854030  test_as_server:TestDownload:278  Thread still running Twisted, daemon: True, instance: <Thread(Twisted, started daemon 140133267416832)>
DEBUG   1461854030  test_as_server:TestDownload:278  Thread still running PoolThread-twisted.internet.reactor-0, daemon: True, instance: <Thread(PoolThread-twisted.internet.reactor-0, started daemon 140132913927936)>
DEBUG   1461854030  test_as_server:TestDownload:278  Thread still running WatchDog, daemon: True, instance: <WatchDog(WatchDog, started daemon 140132961994496)>
INFO    1461854030         _legacy:twisted:154  (TCP Port 45176 Closed)
INFO    1461854030         _legacy:twisted:154  Stopping factory <twisted.web.server.Site instance at 0x7f733c096950>
ERROR   1461854030  test_as_server:TestDownload:138  The reactor was dirty:
ERROR   1461854030  test_as_server:TestDownload:140  >     <DelayedCall 0x7f733c0267a0 [0.706408977509s] called=0 cancelled=0 <lambda>()>
--------------------- >> end captured logging << ---------------------
```

**Comment by lfdversluis** [comment]
Another ref: https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2304/testReport/junit/Tribler.Test.API.test_download/TestDownload/test_download_torrent_from_url/

**Comment by brussee** [comment]
```
I: test_add_torrent_from_url_to_channel_with_description (Tribler.Test.Core.Modules.RestApi.Channels.test_channels_torrents_endpoint.TestModifyChannelTorrentEndpoint) ... 2016-07-04 16:21:10,891 [ERROR] The reactor was dirty:
I: 2016-07-04 16:21:10,894 [ERROR] >     <DelayedCall 0x9792a968 [0.772477149963s] called=0 cancelled=0 LoopingCall<5>(ChannelCommunity.take_step, *(), **{})()>
I: 2016-07-04 16:21:10,897 [ERROR] >     <DelayedCall 0x978549e0 [4.73240303993s] called=0 cancelled=0 LoopingCall<5.0>(ChannelCommunity._periodically_clean_delayed, *(), **{})()>
I: ERROR
I: ======================================================================
I: ERROR: test_add_torrent_from_url_to_channel_with_description (Tribler.Test.Core.Modules.RestApi.Channels.test_channels_torrents_endpoint.TestModifyChannelTorrentEndpoint)
I: ----------------------------------------------------------------------
I: Traceback (most recent call last):
I:   File "/data/data/org.tribler.android/files/lib/python2.7/site-packages/Tribler/Test/test_as_server.py", line 274, in tearDown
I:     super(TestAsServer, self).tearDown(annotate=False)
I:   File "/data/data/org.tribler.android/files/lib/python2.7/site-packages/Tribler/Core/Utilities/twisted_thread.py", line 151, in errback
I:     failure.raiseException()
I:   File "/data/data/org.tribler.android/files/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
I:     current.result = callback(current.result, *args, **kw)
I:   File "/data/data/org.tribler.android/files/lib/python2.7/site-packages/Tribler/Test/test_as_server.py", line 143, in checkReactor
I:     self.assertFalse(delayed_calls, "The reactor was dirty when tearing down the test")
I: AssertionError: The reactor was dirty when tearing down the test
I: ----------------------------------------------------------------------
```

**Comment by brussee** [comment]
This was fixed in #2423.
